src/tiled/tileseteditor.cpp

```diff
@@ -435,21 +436,47 @@ void TilesetEditor::addTiles()
         QPixmap image;
     };
     QVector<LoadedFile> loadedFiles;
-
+    bool addImage = true, dontAskAgain = false;
     for (const QString &file : files) {
-        const QPixmap image(file);
-        if (!image.isNull()) {
-            loadedFiles.append(LoadedFile { file, image });
-        } else {
-            QMessageBox warning(QMessageBox::Warning,
-                                tr("Add Tiles"),
-                                tr("Could not load \"%1\"!").arg(file),
-                                QMessageBox::Ignore | QMessageBox::Cancel,
-                                mMainWindow->window());
-            warning.setDefaultButton(QMessageBox::Ignore);
-
-            if (warning.exec() != QMessageBox::Ignore)
-                return;
+        // If the tile is already in the tileset, warn user and confirm addition
+        const QMap<int, Tile*> addedTiles = mCurrentTilesetDocument->tileset().data()->tiles();
+        for(auto tile : addedTiles) {
+            if(dontAskAgain)
+                break;
```

Comment:
It's alright! Indeed it's not uncommon for bugs to creep in when making "last minute" reorganizations.